resources/js/components/ui/calendar/FullCalendar.tsx

```diff
@@ -54,6 +59,23 @@ const defaultProps = () =>
 
 const PIXELS_PER_HOUR_RANGE = [40, 400] as const;
 
+/**
+ * The state of the calendar persisted in the local storage.
+ *
+ * Warning: Changing this type may break the persistence and the whole application in a browser.
+ * Either make sure the change is backwards compatible (allows reading earlier data),
+ * or bump the version of the persistence.
+ */
+type PersistentState = {
+  readonly mode: Mode;
+  readonly daysSel: readonly [Mode, DaysRange][];
+  readonly resourcesSel: {
+    readonly checkbox: ReadonlySet<string>;
```

Comment:
Może tego nie couplować z UIem? tylko jakoś domenowo (w tej chwili inne wybory są multi- a inne single-choice).

Dodatkowo single-choice wciąż można wrzucić do Array/Set, co jest bardziej backward compatible, a nie jakoś bardzo upierdliwe przy odczytywaniu

Reply:
Próbowałem to pierwotnie nazwać single i multi, tylko że to w sumie bez sensu, bo zobacz że te nazwy radio i checkbox są tu dobrze bardzo osadzone, nawet ResourcesSelector ma je w interface, więc teraz wymyślanie innych nazw akurat w tym miejscu to by tylko zaciemniało.

Natomiast możliwe że to nie są najświetniejsze nazwy... ale to już co najwyżej trzeba zmienić wszędzie, ale wątpię że się to stanie.

No i tablica o tyle nic nie pomaga, że tak czy siak muszę trzymać jedne i drugie dane osobno, więc po prostu daję im ich naturalny format.